atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SerializableTransaction.java

```diff
@@ -880,6 +881,11 @@ private Transaction getReadOnlyTransaction(final long commitTs) {
                 transactionConfig,
                 conflictTracer,
                 tableLevelMetricsController) {
+            @Override
+            protected TransactionScopedCache getCache() {
+                return lockWatchManager.getReadOnlyTransactionScopedCache(SerializableTransaction.this.getTimestamp());
```

Comment:
```suggestion
                return lockWatchManager.getReadOnlyTransactionScopedCache(this.getTimestamp());
```

Reply:
I'm actually not going to do this: you'll notice that this pattern is used in `getCommitTimestamps` too, and I think it makes it _really_ clear which method we mean to call.